lib/js/src/manager/file/filetypes/SdlFile.js

```diff
@@ -223,6 +223,9 @@ class SdlFile {
     clone () {
         const clonedParams = Object.assign({}, this); // shallow copy
 
+        if (this.getFileData() !== null && this.getFileData() !== undefined) {
+            return Object.assign(new SdlFile(this.getName(), this.getType(), Uint8Array.from(this.getFileData()), this.isPersistent()), clonedParams);
```

Comment:
The parameters for Object.assign go as follows: target, source. This means that the second parameter's properties will overwrite whatever is in the first object's properties. In this case, the newly constructed fileData in the first argument is overwritten by the clonedParams, and the deep cloning isn't fully achieved. You may want to check the SdlArtwork clone method for the same reason.